src/java/org/apache/cassandra/service/accord/AccordService.java

```diff
@@ -532,22 +541,21 @@ else if (!tableId.equals(newTableId))
     }
 
     @VisibleForTesting
-    static ReadTimeoutException newBarrierTimeout(TxnId txnId, boolean global)
+    static ReadTimeoutException newBarrierTimeout(TxnId txnId, BarrierType barrierType, boolean isForWrite, Seekables<?, ?> keysOrRanges)
     {
-        return new ReadTimeoutException(global ? ConsistencyLevel.ANY : ConsistencyLevel.QUORUM, 0, 0, false, txnId.toString());
+        return new ReadTimeoutException(barrierType.global ? ConsistencyLevel.ANY : ConsistencyLevel.QUORUM, 0, 0, false, String.format("Timeout waiting on barrier %s / %s / %s; impacted ranges %s", txnId, barrierType, isForWrite ? "write" : "not write", keysOrRanges));
```

Comment:
this was a improvement in CASSANDRA-19833 and was easier to keep when I spun off the debug logic (as its one of the other changes to this class)...  I could split this out if desired

When repair sees these it just logged the txnId, so more detail is just for repair error messages